iocore/net/UnixNet.cc

```diff
@@ -525,7 +531,17 @@ NetHandler::mainNetEvent(int event, Event *e)
       close_UnixNetVConnection(vc, trigger_event->ethread);
     else if (vc->read.enabled && vc->read.triggered)
       vc->net_read_io(this, trigger_event->ethread);
-    else if (!vc->read.enabled) {
+    else if (vc->read.error) {
+      int err = 0, errlen = sizeof(int);
+      if (getsockopt(vc->con.fd, SOL_SOCKET, SO_ERROR, &err, (socklen_t *)&errlen) == -1) {
+        err = errno;
+      }
+      if (err != EAGAIN && err != EINTR) {
+        vc->readSignalError(this, err);
```

Comment:
That makes sense :) I've cleaned that up (left it as a separate commit for review-- although before merging I'll squash it). I did test that its still fixing my bug-- and it is, so I think we are set :)